packages/openid4vc/src/shared/utils.ts

```diff
@@ -34,61 +38,95 @@ export function getSupportedJwaSignatureAlgorithms(agentContext: AgentContext):
+    const { iss, sub, aud, exp, nbf, iat, jti, ...additionalClaims } = jwt.payload
+
+    const jwsService = agentContext.dependencyManager.resolve(JwsService)
+
+    if (jwtIssuer.method === 'did') {
+      const key = await getKeyFromDid(agentContext, jwtIssuer.didUrl)
+      const jws = await jwsService.createJwsCompact(agentContext, {
+        protectedHeaderOptions: { alg: jwtIssuer.alg, ...jwt.header },
+        payload: new JwtPayload({ ...jwt.payload, additionalClaims }),
+        key,
+      })
+
+      return jws
+    } else if (jwtIssuer.method === 'jwk') {
+      const key = getJwkFromJson(jwtIssuer.jwk).key
+      const jws = await jwsService.createJwsCompact(agentContext, {
+        protectedHeaderOptions: jwt.header as JwsProtectedHeaderOptions,
+        payload: new JwtPayload({ ...jwt.payload, additionalClaims }),
+        key,
+      })
+
+      return jws
+    } else if (jwtIssuer.method === 'x5c') {
+      const key = X509Service.getLeafCertificate(agentContext, { certificateChain: jwtIssuer.x5c }).publicKey
 
-  return {
-    signature: async (data: string | Uint8Array) => {
-      if (typeof data !== 'string') throw new CredoError("Expected string but received 'Uint8Array'")
-      const signedData = await agentContext.wallet.sign({
-        data: TypedArrayEncoder.fromString(data),
+      const jws = await jwsService.createJwsCompact(agentContext, {
+        protectedHeaderOptions: jwt.header as JwsProtectedHeaderOptions,
+        payload: new JwtPayload({ ...jwt.payload, additionalClaims }),
         key,
       })
 
-      const signature = TypedArrayEncoder.toBase64URL(signedData)
-      return signature
-    },
-    alg: alg as unknown as SigningAlgo,
-    did,
-    kid,
+      return jws
+    }
+
+    throw new Error(`Unsupported jwt issuer method '${jwtIssuer.method}'`)
   }
 }
 
-export function getSphereonDidResolver(agentContext: AgentContext) {
-  return {
-    resolve: async (didUrl: string) => {
-      const didsApi = agentContext.dependencyManager.resolve(DidsApi)
-      const result = await didsApi.resolve(didUrl)
-
-      return {
-        ...result,
-        didDocument: result.didDocument?.toJSON() as DIDDocument,
-      }
-    },
+export async function openIdTokenIssuerToJwtIssuer(
+  agentContext: AgentContext,
+  openId4VcTokenIssuer: OpenId4VcJwtIssuer
+): Promise<JwtIssuer> {
+  if (openId4VcTokenIssuer.method === 'did') {
+    const key = await getKeyFromDid(agentContext, openId4VcTokenIssuer.didUrl)
+    const _alg = getJwkClassFromKeyType(key.keyType)?.supportedSignatureAlgorithms[0]
+    if (!_alg) throw new CredoError(`No supported signature algorithms for key type: ${key.keyType}`)
+
+    return {
+      method: openId4VcTokenIssuer.method,
+      didUrl: openId4VcTokenIssuer.didUrl,
+      alg: _alg as unknown as SigningAlgo,
+    }
+  } else if (openId4VcTokenIssuer.method === 'x5c') {
+    return {
+      ...openId4VcTokenIssuer,
+      clientIdScheme: 'x509_san_dns',
```

Comment:
In other code I saw you also handled `x509_san_uri`. Shouldn't we handle that here as well?